Source/WebCore/html/parser/HTMLDocumentParser.cpp

```diff
@@ -592,6 +594,15 @@ void HTMLDocumentParser::notifyFinished(PendingScript& pendingScript)
     ASSERT(m_scriptRunner);
     ASSERT(!isExecutingScript());
     if (isStopping()) {
+        // If we're currently in a microtask checkpoint, schedule end() as a regular task.
+        // This ensures it runs after ALL microtasks (including any created during execution) complete.
+        RefPtr document = this->document();
+        if (document->eventLoop().microtaskQueue().isPerformingCheckpoint()) {
```

Comment:
Is this check really necessary? It's somewhat arbitrary to delay the end of parsing only when we're inside a micro-task.

Reply:
I tried that and it breaks many tests and assertion failures. See this build. https://github.com/WebKit/WebKit/pull/49910#issuecomment-3230457669